specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json

```diff
@@ -1012,6 +1012,295 @@
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/enterpriseChannels/{resourceName}": {
+      "put": {
+        "tags": [
+          "Enterprise Channel"
+        ],
+        "description": "Creates an Enterprise Channel.",
+        "operationId": "EnterpriseChannels_Create",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options" : {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Create Enterprise Channel": {
+            "$ref": "./examples/CreateEnterpriseChannel.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            },
+            "description": "The parameters to provide for the new Enterprise Channel."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "If resource is created successfully or already existed, the service should return 200 (OK).",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (Created). Execution to continue asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Enterprise Channel"
+        ],
+        "description": "Updates an Enterprise Channel.",
+        "operationId": "EnterpriseChannels_Update",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options" : {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Update Enterprise Channel": {
+            "$ref": "./examples/UpdateEnterpriseChannel.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
```

Comment:
Hi @NickEricson - last question:
Are you sure you wish to flatten out the body parameter here? It's not wrong - but does result in rather a large generated method signature:
https://github.com/Azure/azure-sdk-for-python/pull/3151/files#diff-c30e6c4ddd2d3e8f68fd313d4757fe05R327

Reply:
We are using it on our other objects as well and would like to be consistent.